hal/armv7a/exceptions.c

```diff
@@ -232,3 +243,85 @@ void _hal_exceptionsInit(void)
 	exceptions.abortHandler = exceptions_defaultHandler;
 	exceptions.defaultHandler = exceptions_defaultHandler;
 }
+
+
+cpu_context_t *hal_excToCpuCtx(exc_context_t *ctx)
+{
+	return &ctx->cpuCtx;
+}
+
+
+void hal_coredumpGRegset(void *buff, cpu_context_t *ctx)
+{
+	u32 *regs = (u32 *)buff;
+	*(regs++) = ctx->r0;
+	*(regs++) = ctx->r1;
+	*(regs++) = ctx->r2;
+	*(regs++) = ctx->r3;
+	*(regs++) = ctx->r4;
+	*(regs++) = ctx->r5;
+	*(regs++) = ctx->r6;
+	*(regs++) = ctx->r7;
+	*(regs++) = ctx->r8;
+	*(regs++) = ctx->r9;
+	*(regs++) = ctx->r10;
+	*(regs++) = ctx->fp;
+	*(regs++) = ctx->ip;
+	*(regs++) = ctx->sp;
+	*(regs++) = ctx->lr;
+	*(regs++) = ctx->pc;
+	*(regs++) = ctx->psr;
+}
+
+
+void hal_coredumpThreadAux(void *buff, cpu_context_t *ctx)
+{
+	const char ARMVFP_NAME[] = "LINUX";
+	Elf32_Nhdr nhdr;
+	nhdr.n_namesz = sizeof(ARMVFP_NAME);
```

Comment:
With `nul` char ?

Reply:
Yes, that's how it's done in linux kernel